### PR TITLE
feat: generate changelog from conventional commits

### PR DESCRIPTION
**Pull request by Insomniak47**
Added initial parsing and grouping of categories for the changelog

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -28,8 +31,8 @@ async fn main() -> Result<(), Error> {
     let client = reqwest::Client::new();
     let context = github::Context::new(
         &client,
-        &matches.get_one::<String>("owner").unwrap(),
-        &matches.get_one::<String>("repo").unwrap(),
+        matches.get_one::<String>("owner").unwrap(),
```
Comment (ttd2089) [inline src/main.rs:34]: Why did we drop the `&`?

File: src/main.rs [hunk 2]
```diff
@@ -41,10 +44,68 @@ async fn main() -> Result<(), Error> {
     let comparison = context
         .compare_commits(
             &release.target_commitish,
-            &matches.get_one::<String>("target").unwrap(),
+            matches.get_one::<String>("target").unwrap(),
```
Comment (ttd2089) [inline src/main.rs:47]: Why did we drop the `&`?

File: src/main.rs [hunk 3]
```diff
@@ -41,10 +44,68 @@ async fn main() -> Result<(), Error> {
...
+    match matches.subcommand() {
+        Some(("version", _)) => println!("You typed version :D"),
+        Some(("changelog", _)) => print_changelog(&commits),
+        _ => unreachable!("Oh nooo"),
+    };
+
+    // let commit = conventional::Commit::new("strang").expect("Should be here");
 
-    println!("{:?}", comparison);
+    // println!("{:?}", comparison);
```
Comment (ttd2089) [inline src/main.rs]:
```suggestion
```

File: src/main.rs [hunk 4]
```diff
@@ -41,10 +44,68 @@ async fn main() -> Result<(), Error> {
...
+        .filter(is_relevant_commit)
+        .collect::<Vec<_>>();
+
+    match matches.subcommand() {
+        Some(("version", _)) => println!("You typed version :D"),
+        Some(("changelog", _)) => print_changelog(&commits),
+        _ => unreachable!("Oh nooo"),
+    };
+
+    // let commit = conventional::Commit::new("strang").expect("Should be here");
```
Comment (ttd2089) [inline src/main.rs]:
```suggestion
```

File: src/main.rs (hunk 1 above)
Comment (Insomniak47) [inline src/main.rs:34]: clippy lint here was:
```
warning: this expression creates a reference which is immediately dereferenced by the compiler
  --> src\main.rs:34:9
   |
34 |         &matches.get_one::<String>("owner").unwrap(),
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: change this to: `matches.get_one::<String>("owner").unwrap()`
   |
   = note: `#[warn(clippy::needless_borrow)]` on by default
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_borrow
```

File: src/main.rs [hunk 5]
```diff
@@ -41,10 +44,68 @@ async fn main() -> Result<(), Error> {
...
+        categories
+            .entry(key)
+            .and_modify(|x| x.push(commit))
+            .or_insert_with(Vec::new);
+    }
+
+    for (category, title) in [
+        ("breaking", "breaking changes"),
+        ("feat", "features"),
+        ("fix", "fixes"),
```
Comment (ttd2089) [inline src/main.rs]:
```suggestion
        ("breaking", "BREAKING CHANGES"),
        ("feat", "Features"),
        ("fix", "Bug Fixes"),
```

File: src/main.rs [hunk 6]
```diff
@@ -41,10 +44,68 @@ async fn main() -> Result<(), Error> {
...
+    }
+}
+
+fn print_category(
+    hash_map: &HashMap<String, Vec<&conventional::Commit>>,
+    category_name: &str,
+    title: &str,
+) {
+    if let Some(breaking) = hash_map.get(category_name) {
+        println!("{} stuff!\n", title);
```
Comment (ttd2089) [inline src/main.rs]:
```suggestion
        println!("{}:\n", title);
```

**Review by ttd2089 — CHANGES_REQUESTED** [review]

File: src/main.rs (hunk 1 above)
Comment (ttd2089) [reply src/main.rs:34]: oh haha

**Review by ttd2089 — APPROVED** [review]

**Pull request merged**